### PR TITLE
KOGITO-8337 Setup e2e nightlies

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-8337

- Jenkins nightly uses minikube
- Removed KinD GHA and use only minikube
- updated `go.mod` due to issue with the install of `setup-envtest` binary

### CODE REVIEW

File: .ci/jenkins/Jenkinsfile.deploy [hunk 1]
```diff
@@ -51,6 +51,9 @@ pipeline {
 
                     // Verify we have the connectivity to Openshift
                     openshift.loginOpenshift()
+
+                    // Fix for rhel8 nodes
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.deploy:55]: I sent a PR to update jenkins node
I will remove this once the new node with the correction will be live.

**Review by ricardozanini — APPROVED** [review]

**Review by desmax74 — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -21,56 +22,53 @@ env:
...
-      id: kind
-      uses: container-tools/kind-action@v2
-      with:
-        version: v0.17.0
-        node_image: kindest/node:${{ env.KUBERNETES_VERSION }}
-        kubectl_version: ${{ env.KUBERNETES_VERSION }}
-        wait: 120s
+      - name: Build operator image
+        run: |
+          make container-build BUILDER=docker IMG=${{ env.OPERATOR_IMAGE_NAME }}
```
Comment (spolti) [inline .github/workflows/e2e.yml:42]: can't we switch to podman?

**Review by spolti — APPROVED** [review]

File: .github/workflows/e2e.yml (hunk 2 above)
Comment (radtriste) [reply .github/workflows/e2e.yml:42]: well I need to make sure it works. let's try !

**Review by davidesalerno — APPROVED** [review]
lgtm

File: .ci/jenkins/Jenkinsfile.deploy (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.deploy]: updated

**Pull request merged**